autoload/ale/lsp/response.vim

```diff
@@ -28,7 +28,7 @@ function! ale#lsp#response#ReadDiagnostics(response) abort
     for l:diagnostic in a:response.params.diagnostics
         let l:severity = get(l:diagnostic, 'severity', 0)
         let l:loclist_item = {
-        \   'text': l:diagnostic.message,
+        \   'text': substitute(l:diagnostic.message, "\n", ' ', 'g'),
```

Comment:
```suggestion
        \   'text': substitute(l:diagnostic.message, '\(\r\n\|\n\|\r\)', ' ', 'g'),
```

I think you should handle Windows style CRLF newlines too. For example:

```vim
:echo substitute("one\ntwo\r\nthree\rfour", '\(\r\n\|\n\|\r\)', ' ', 'g')
```

The string for the regex has to be single quoted here. Add tests for `\r` by itself and `\r\n`.